os/test/src-jvm/PathTestsCustomFilesystem.scala

```diff
@@ -175,6 +176,50 @@ object PathTestsCustomFilesystem extends TestSuite {
           assert(os.exists(p / "file.txt"))
         }
       }
+      test("copyAndMergeToRootDirectoryWithCreateFolders") {
+        withCustomFs { fileSystem =>
+          val root = os.root("/", fileSystem)
+          val file = root / "test" / "dir" / "file.txt"
+          os.write(file, "Hello World")
+          os.copy(root / "test" / "dir", root, createFolders = true, mergeFolders = true)
+          assert(os.read(root / "file.txt") == "Hello World")
+          assert(os.exists(root / "file.txt"))
+        }
+      }
+      test("moveToRootDirectoryWithCreateFolders") {
+        withCustomFs { fileSystem =>
+          // This should fail. Just test that it doesn't throw PathError.AbsolutePathOutsideRoot.
+          val root = os.root("/", fileSystem)
+          try {
+            os.move(root / "test" / "dir", root, createFolders = true)
+          } catch {
+            case e: PathError.AbsolutePathOutsideRoot.type => throw e
+            case NonFatal(_) => ()
+          }
```

Comment:
You could also use an `intercept` here, to declare the concrete expected exception.

https://github.com/com-lihaoyi/utest?tab=readme-ov-file#intercept